packages/transactions/src/__tests__/new-compile-transaction-test.ts

```diff
@@ -0,0 +1,55 @@
+import '@solana/test-matchers/toBeFrozenObject';
+
+import { Address } from '@solana/addresses';
+
+import { CompiledMessage, compileMessage } from '../message';
+import { compileTransactionMessage } from '../new-compile-transaction';
+import { getCompiledMessageEncoder } from '../serializers';
+
+jest.mock('../message');
+jest.mock('../serializers/message');
+
+type TransactionMessage = Parameters<typeof compileTransactionMessage>[0];
+
+describe('compileTransactionMessage', () => {
+    const mockAddressA = '1111' as Address;
+    const mockAddressB = '2222' as Address;
+    const mockCompiledMessage = {
+        header: {
+            numSignerAccounts: 2,
+        },
+        staticAccounts: [mockAddressA, mockAddressB],
+    } as CompiledMessage;
+    const mockCompiledMessageBytes = new Uint8Array(Array(100)).fill(1);
+    beforeEach(() => {
+        (compileMessage as jest.Mock).mockReturnValue(mockCompiledMessage);
+        (getCompiledMessageEncoder as jest.Mock).mockReturnValue({
+            encode: jest.fn().mockReturnValue(mockCompiledMessageBytes),
+        });
+    });
+
+    it('compiles the transaction message', () => {
```

Comment:
Sorrrrrrrt of more like…
```suggestion
    it('compiles the supplied `TransactionMessage` and sets the `messageBytes` property to the result', () => {
```